test/integration/scheduler/priorities_test.go

```diff
@@ -172,3 +173,61 @@ func TestPodAffinity(t *testing.T) {
 	}
 	t.Errorf("Pod %v got scheduled on an unexpected node: %v.", podName, pod.Spec.NodeName)
 }
+
+// TestImageLocality verifies that the scheduler's image locality priority function
+// works correctly
+func TestImageLocality(t *testing.T) {
+	context := initTest(t, "image-locality")
+	defer cleanupTest(t, context)
+	// Add a few nodes.
+	nodes, err := createNodes(context.clientSet, "testnode", nil, 5)
+	if err != nil {
+		t.Fatalf("Cannot create nodes: %v", err)
+	}
+	// Add an image to one of the nodes and we use mnist-tpu as the test image(s), which
+	// has relatively large image size
+	nodeWithImages := nodes[1]
+	images := []imageutils.ImageConfig{
+		imageutils.MnistTpu,
+	}
+	if err = testutils.AddImagesToNode(context.clientSet, nodeWithImages.Name, images); err != nil {
```

Comment:
Please also add an un-used images to other one node.

Reply:
Do you mean for each of the other 4 test nodes we should add a random image (any image except for mnist-tpu)? Thanks!

Reply:
I think it's better to have a `createNodesWithImages(...) `, WDYT?

Reply:
Yes, this is certainly the cleaner alternative (createNodeWithImages as opposed to indirectly patching node status) - however I think the current choice of reusing createNode + calling addImagesToNode combination is more generic (similar to the existing createNode + addLabelToNode combination) and adhering to the image pulling logic.. - do you think we should change to the alternative?

Reply:
> Do you mean for each of the other 4 test nodes we should add a random image (any image except for mnist-tpu)?

one node is enough; so we can test: no images node, matched images node and dis-matched images node.